migrate_cpp/rewriter_test.cpp

```diff
@@ -0,0 +1,135 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "migrate_cpp/rewriter.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "clang/AST/ASTConsumer.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendAction.h"
+#include "clang/Tooling/Tooling.h"
+
+namespace Carbon::Testing {
+namespace {
+
+// Represents C++ source code with at most one region enclosed in $[[...]]$ as
+// an annotated range.
+class Annotations {
+ public:
+  Annotations(llvm::StringRef annotated_source) {
+    size_t index = annotated_source.find("$[[");
+    if (index == llvm::StringRef::npos) {
+      source_code = std::string(annotated_source);
+      return;
+    }
+    start = index;
+    end = annotated_source.find("]]$", index);
+    CARBON_CHECK(end != llvm::StringRef::npos)
+        << "Found `$[[` but no matching `]]$`";
+    source_code = (llvm::Twine(annotated_source.substr(0, start)) +
+                   annotated_source.substr(start + 3, end - start - 3) +
+                   annotated_source.substr(end + 3))
+                      .str();
+    // Update `end` so that it is relative to the unannotated source (which
+    // means three characters earlier due to the `$[[` being removed.
+    end -= 3;
+  }
+
+  // Returns a view into the unannotated source.
+  llvm::StringRef source() const { return source_code; }
+
+  // Returns the offsets in the file representing the annotated range if they
+  // exist and `{0, std::numeric_limits<size_t>::max()}` otherwise.
+  std::pair<size_t, size_t> range() const { return std::pair(start, end); }
+
+ private:
+  std::string source_code;
+  size_t start = 0;
+  size_t end = std::numeric_limits<size_t>::max();
```

Comment:
```suggestion
  std::string source_code_;
  size_t start_ = 0;
  size_t end_ = std::numeric_limits<size_t>::max();
```

Underscores for private data members

Reply:
Missed these. Thanks. Done.